src/main/java/lotto/controller/LottoGameController.java

```diff
@@ -0,0 +1,83 @@
+package lotto.controller;
+
+import lotto.dto.FinalResultsDto;
+import lotto.dto.LottosDto;
+import lotto.service.puchase.LottoPurchaseService;
+import lotto.service.result.LottoResultService;
+import lotto.view.input.InputView;
+import lotto.view.output.OutputView;
+
+public class LottoGameController {
+    private final LottoPurchaseService lottoPurchaseServiceImpl;
+    private final LottoResultService lottoResultServiceImpl;
+    private final InputView consoleInputView;
+    private final OutputView consoleOutputView;
+
+    public LottoGameController(OutputView consoleOutputView
+            , InputView consoleInputView
+            , LottoPurchaseService lottoPurchaseServiceImpl
+            , LottoResultService lottoResultServiceImpl) {
+
+        this.lottoPurchaseServiceImpl = lottoPurchaseServiceImpl;
+        this.lottoResultServiceImpl = lottoResultServiceImpl;
+        this.consoleInputView = consoleInputView;
+        this.consoleOutputView = consoleOutputView;
+
+    }
+
+    public void run() {
+
+        LottosDto lottosDto = executeLottoPurchase();
+        consoleOutputView.outputPurchaseLottoList(lottosDto);
+
+        handleWinningNumbersInput();
+        handleBonusNumberInput();
+
+        FinalResultsDto finalResultsDto = lottoResultServiceImpl.getFinalResultsDto(lottosDto);
+        consoleOutputView.outputFinalResult(finalResultsDto);
+
+    }
+
+    private LottosDto executeLottoPurchase() {
+        while (true) {
+            try {
+                String rawPurchaseAmount = consoleInputView.inputPurchaseAmount();
+                lottoPurchaseServiceImpl.purchaseLottos(rawPurchaseAmount);
+
+                break;
+
+            } catch (IllegalArgumentException e) {
+                consoleOutputView.outputErrorMessage(e.getMessage());
+            }
+        }
+        return lottoPurchaseServiceImpl.getLottosDto();
```

Comment:
```suggestion
        return lottoPurchaseServiceImpl.getLottosDto();

            } catch (IllegalArgumentException e) {
                consoleOutputView.outputErrorMessage(e.getMessage());
            }
        }
```
break문 없이 리턴 문을 바로 사용해도 좋을 거 같습니다!

Reply:
구글링을 통해 찾은 포맷을 크게 고민없이 썼던 것 같네요. 의미를 보았을 때 return이 더 나은 것 같습니다. 세심한 지적 감사합니다